lewis_emulators/aldn1000/interfaces/stream_interface.py

```diff
@@ -0,0 +1,166 @@
+from lewis.adapters.stream import StreamInterface, Cmd
+
+from lewis_emulators.utils.command_builder import CmdBuilder
+from lewis_emulators.utils.constants import STX, ETX
+from lewis_emulators.utils.replies import conditional_reply
+
+if_input_error = conditional_reply('input_correct', STX + "01I?NA" + ETX)
+if_connected = conditional_reply("connected")
+
+
+class Aldn1000StreamInterface(StreamInterface):
+
+    in_terminator = "\r"
+
+    directions = 'INF|WDR|REV'
+    status_modes = 'I|W|S|P|T|U|A?R|A?S|A?T|A?E|A?O'
+
+    def __init__(self):
+        super(Aldn1000StreamInterface, self).__init__()
+
+        self.commands = {
+                CmdBuilder(self.get_diameter).int().escape('DIA').eos().build(),
+                CmdBuilder(self.set_diameter).int().escape('DIA').float().eos().build(),
+                CmdBuilder(self.get_volume).int().escape('VOL').eos().build(),
+                CmdBuilder(self.set_volume).int().escape('VOL').float().eos().build(),
+                CmdBuilder(self.get_direction).int().escape('DIR').eos().build(),
+                CmdBuilder(self.set_direction).int().escape('DIR').arg(self.directions).eos().build(),
+                CmdBuilder(self.get_rate).int().escape('RAT').eos().build(),
+                CmdBuilder(self.set_rate).int().escape('RAT').float().string().eos().build(),
+                CmdBuilder(self.get_program_function).int().escape('FUN').eos().build(),
+                CmdBuilder(self.get_volume_dispensed).int().escape('DIS').eos().build(),
+                CmdBuilder(self.clear_volume).int().escape('CLD').string().eos().build(),
+                CmdBuilder(self.set_pump).int().arg('STP|RUN').eos().build(),
+                CmdBuilder(self.get_status).int().eos().build(),
+        }
+
+    @if_input_error
+    def basic_get_response(self, address, data=None, units=None):
+        # The device requires very specific formatting(0000. - 1000.) for floats, so we restrict the float output to
```

Comment:
Created full docstring for the `format_data` method and added comment on `get_basic_response` so that users know to format data passed to this method first.